frontend/src/entry/model/settings/EntryRolesInfo.tsx

```diff
@@ -54,36 +54,105 @@ export default function EntryRolesInfo({ entry }: EntryRolesInfoProps) {
-                  <Typography variant='caption'>
-                    {`The following roles are specified by this ${toSentenceCase(entry.kind)}'s schema`}
-                  </Typography>
-                </Box>
-                {schemaRolesList}
-              </Stack>
-            </Grid>
-          </Grid>
+          <Stack spacing={2}>
+            <Stack spacing={1}>
+              <Typography component='h3' variant='h6' fontWeight='bold'>
+                General Permissions
+              </Typography>
+              <TableContainer component={Paper} sx={{ maxHeight: 250 }}>
+                <Table sx={{ minWidth: 650 }} stickyHeader size='small'>
+                  <TableHead>
+                    <TableRow>
+                      <TableCell>Permission</TableCell>
+                      <TableCell align='right'>Owner</TableCell>
+                      <TableCell align='right'>Contributor</TableCell>
+                      <TableCell align='right'>Consumer</TableCell>
+                      <TableCell align='right'>Release Reviewer</TableCell>
+                      <TableCell align='right'>Access Reviewer</TableCell>
+                    </TableRow>
+                  </TableHead>
+                  <TableBody>
+                    {rows.map((row) => (
+                      <TableRow key={row.permission} sx={{ '&:last-child td, &:last-child th': { border: 0 } }}>
+                        <TableCell component='th' scope='row'>
+                          {row.permission}
+                        </TableCell>
+                        <TableCell align='center'>{row.roles.includes('owner') ? <Done /> : <></>}</TableCell>
+                        <TableCell align='center'>{row.roles.includes('contributor') ? <Done /> : <></>}</TableCell>
+                        <TableCell align='center'>{row.roles.includes('consumer') ? <Done /> : <></>}</TableCell>
+                        <TableCell align='center'>{row.roles.includes('releaseReviewer') ? <Done /> : <></>}</TableCell>
+                        <TableCell align='center'>{row.roles.includes('accessReviewer') ? <Done /> : <></>}</TableCell>
+                      </TableRow>
+                    ))}
+                  </TableBody>
+                </Table>
+              </TableContainer>
+            </Stack>
+            <Stack spacing={1}>
+              <Box>
+                <Typography component='h3' variant='h6' fontWeight='bold'>
+                  Dynamic Roles
+                </Typography>
+                <Typography variant='caption'>
+                  {`The following roles are specified by this ${toSentenceCase(entry.kind)}'s schema.`}
+                </Typography>
+              </Box>
+              {schemaRolesList}
+            </Stack>
+          </Stack>
         </Stack>
       )}
     </>
   )
 }
+
+const rows = [
+  {
+    permission: 'View private model',
+    roles: ['owner', 'contributor', 'consumer', 'releaseReviewer', 'accessReviewer'],
```

Comment:
Would be great if we could have a `RoleValue` "enum" for all of these instead of having to make sure we don't typo any of these string arrays. Feel free to rename if you can think of anything better!